src/test/java/pages/BasePage.java

```diff
@@ -0,0 +1,38 @@
+package pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
+
+public class BasePage {
+
+    WebDriver driver;
+    WebDriverWait wait;
+    Actions actions;
+
+    public BasePage(WebDriver givenDriver){
+        driver = givenDriver;
+        wait = new WebDriverWait(driver, Duration.ofSeconds(5));
+        //actions = new Actions(driver);
+    }
```

Comment:
We should initialize page objects using initElements() method from PageFactory Class as below, Once we call initElements() method, all elements will get initialized.
Constructor for Base page class as below:


```
    public BasePage(WebDriver givenDriver){
        driver = givenDriver;
        wait = new WebDriverWait(driver, Duration.ofSeconds(5));
        PageFactory.initElements(driver, this);
    }
```